### PR TITLE
feat: Show mention suggestion popup (WEBAPP-5462)

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/view_model/content/InputBarViewModel.js [hunk 1]
```diff
@@ -226,6 +214,11 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
     );
   }
 
+  addMention(userEntity) {
+    // TODO
```
Comment (atomrc) [inline app/script/view_model/content/InputBarViewModel.js:220]: This will be done in another PR

File: app/script/components/mentionSuggestions.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    });
+
+    this.suggestions.subscribe(suggestions => {
+      const shouldBeVisible = suggestions.length > 0;
+      if (shouldBeVisible && !this.isVisible()) {
+        this.initList();
+      } else if (!shouldBeVisible && this.isVisible()) {
+        this.teardownList();
+      }
+      this.isVisible(shouldBeVisible);
```
Comment (AndyLnd) [inline app/script/components/mentionSuggestions.js]: `this.selectedSuggestionIndex` should be updated to stay in the bounds of the array length.

File: app/script/components/mentionSuggestions.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.MentionSuggestions = class MentionSuggestions {
+  constructor(params) {
+    this.isVisible = ko.observable(false);
+    this.onSelectionValidated = params.onSelectionValidated || (() => {});
+    this.suggestions = params.suggestions;
+    this.targetInputSelector = params.targetInput;
```
Comment (AndyLnd) [inline app/script/components/mentionSuggestions.js]: please name the param also `targetInputSelector`

File: app/script/components/mentionSuggestions.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    if (action) {
+      action();
+      keyboardEvent.preventDefault();
+      keyboardEvent.stopPropagation();
+    }
+  }
+
+  moveSelection(delta) {
+    const currentIndex = this.selectedSuggestionIndex();
+    const newIndex = Math.max(Math.min(currentIndex + delta, this.suggestions().length - 1), 0);
```
Comment (AndyLnd) [inline app/script/components/mentionSuggestions.js]: we have `z.util.NumberUtil.clamp` for that
